src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1153,72 @@ const isMobileFieldUpdateAllowed = (
     },
   )
 }
+
+export const createTwilioCredentials = (
+  originalForm: IPopulatedForm,
+  twilioCredentials: TwilioCredentials,
+  formId: string,
+): ResultAsync<SecretsManager.Types.CreateSecretRequest, ApplicationError> => {
+  const msgSrvcName = `formsg/${process.env.FORMSG_SDK_MODE}/form/${formId}/twilio`
+
+  // TO DO: Add error handling to undo MongoDB
+  void updateForm(originalForm, { msgSrvcName })
+
+  const body: SecretsManager.Types.CreateSecretRequest = {
+    Name: msgSrvcName,
+    SecretString: JSON.stringify(twilioCredentials),
+  }
+
+  let reqError: AWSError | null
+  const request = secretsManager.createSecret(body, (err: AWSError, data) => {
+    if (err) {
+      logger.error({
+        message: 'Error when creating Secret',
+        meta: {
+          action: 'Create Secret in SecretsManager',
+          data,
+        },
+        error: err,
+      })
+      reqError = err
+    }
+  })
+  request.send()
+
+  return ResultAsync.fromPromise(Promise.resolve(body), () => {
```

Comment:
actually since the secretsManager functions, `createSecret`, `putSecretValue` and `send` (which returns void) are synchronous functions, which means if we do this will still need to wrap the lambda result in a Promise.resolve (...) to use ResultAsync.fromPromise. Should i still go ahead with that?

There dosent seem to be a Promise variant for these functions either :(

Reply:
I think the Application error can be thrown from the createSecret / putSecretValue callback though

Reply:
I don't think they are synchronous, are they? since there is for sure a network call to AWS 😲 . 

Might want to try the promise version?

```ts
const result = await secretsManager.createSecret(body).promise()
```

We can also inline that in the ResultAsync wrapper:

```ts
return ResultAsync.fromPromise(secretsManager.createSecret(body).promise(), (error) => {...})
```

Not tested, not sure if it works like this, but it should?

see: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/using-promises.html

Reply:
Ah ok, was looking at the request.send() docs and didn't see that anything async there. Will check this out and fit it thanks :D